scripts/check_golint.sh

```diff
@@ -4,4 +4,4 @@
 # Run the linter on everything except generated code
 set -euo pipefail
 
-golint -set_exit_status $(GO111MODULE=off go list ./... | grep -v '/mocks')
+golint -set_exit_status $(GO111MODULE=auto go list ./... | grep -v '/mocks')
```

Comment:
Thanks for this change. I believe we have now started to build for go1.17.

Go1.17 will ignore this flag: https://go.dev/blog/go116-module-changes.
I believe: https://github.com/gomods/athens/pull/815 was an explicit choice though to make the CI faster

I am not sure if this will work anymore. Maybe we should just remove it: cc @marwan-at-work ?

Reply:
Yes I think it's worth removing. Also, do we even have `/mock`? I wonder if we can do `golinst ./...` or `golint go list ./...`

Reply:
I'll do it right and raise another PR!

Reply:
I checked `golint ./...` vs. `golint $(go list ./...)`... I am struggling to prove it definitively prove that the former is linting all the files covered by the latter. The `go list` subshell command version takes noticeably longer, but that's not to say, necessarily, that the `golint` command isn't examining all the correct files. Any input on that much appreciated.

Reply:
Thank you for doing it right. I just noticed that even golint is deprecated. 
https://github.com/golang/lint
Would you be able to update the code to use `staticcheck` and `go vet`

We already use govet
https://github.com/gomods/athens/blob/fd54babfae1d5fea4f089789e27b23956532f84c/.drone.yml#L32

I would probably just start with govet, open an issue and then do a separate PR for staticcheck. I hope this helps.